app/models/manageiq/providers/amazon/inventory/parser/cloud_manager.rb

```diff
@@ -317,29 +317,51 @@ def hardware_disks(persister_hardware, instance, flavor)
 
   def flavors
     collector.flavors.each do |flavor|
-      persister.flavors.find_or_build(flavor[:name]).assign_attributes(
-        :name                     => flavor[:name],
-        :description              => flavor[:description],
-        :enabled                  => !flavor[:disabled],
-        :cpus                     => flavor[:vcpu],
-        :cpu_cores                => 1,
-        :memory                   => flavor[:memory],
-        :supports_32_bit          => flavor[:architecture].include?(:i386),
-        :supports_64_bit          => flavor[:architecture].include?(:x86_64),
-        :supports_hvm             => flavor[:virtualization_type].include?(:hvm),
-        :supports_paravirtual     => flavor[:virtualization_type].include?(:paravirtual),
-        :block_storage_based_only => flavor[:ebs_only],
-        :cloud_subnet_required    => flavor[:vpc_only],
-        :ephemeral_disk_size      => flavor[:instance_store_size],
-        :ephemeral_disk_count     => flavor[:instance_store_volumes]
+      persister.flavors.find_or_build(flavor['instance_type']).assign_attributes(
+        :name                     => flavor['instance_type'],
+        :description              => flavor['instance_type'],
+        :enabled                  => true,
+        :cpus                     => flavor.dig('v_cpu_info', 'default_v_cpus'),
+        :cpu_cores                => flavor.dig('v_cpu_info', 'default_cores') || 1,
+        :memory                   => flavor.dig('memory_info', 'size_in_mi_b')&.megabytes,
+        :supports_32_bit          => flavor.dig('processor_info', 'supported_architectures').include?('i386'),
+        :supports_64_bit          => flavor.dig('processor_info', 'supported_architectures').include?('x86_64'),
+        :supports_hvm             => true,
+        :supports_paravirtual     => supports_paravirtual?(flavor),
+        :block_storage_based_only => flavor.dig('supported_root_device_types').any? { |device| device != 'ebs' },
+        :ephemeral_disk_size      => flavor.dig('instance_storage_info', 'total_size_in_gb')&.gigabytes || 0,
+        :ephemeral_disk_count     => flavor.dig('instance_storage_info', 'disks')&.size || 0,
+        :publicly_available       => true,
+        :cloud_subnet_required    => vpc_only?(flavor)
```

Comment:
Can you move this back up under block_storage_based_only where it was previously?

Reply:
Done.